File: python/email_validation_test.py
from onboarding_buddy_client.api.validation_api import ValidationApi
from onboarding_buddy_client.models.email_address_request_m import EmailAddressRequestM
from onboarding_buddy_client import Configuration, ApiClient
from dotenv import load_dotenv
import os

load_dotenv()

# Configure the client with authentication headers
config = Configuration(
    api_key={
        "AppKey": os.getenv("OB_APP_KEY"),
        "ApiKey": os.getenv("OB_API_KEY"),
        "ApiSecret": os.getenv("OB_API_SECRET")
    }
)

# Initialize the API client
api_client = ApiClient(configuration=config)
validation_api = ValidationApi(api_client)

# Submit the email validation request
try:
    # Create an email validation request
    request = EmailAddressRequestM(
        email_address="email@domain.com"
    )

    response = validation_api.email(email_address_request_m=request)
    
    print(f"Email Address: {response.email_address}")
    print(f"Email Status: {response.email_status}")
    print(f"Free Email: {response.free_email}")
    print(f"Domain: {response.domain}")
    print(f"MX Record: {response.mx_record}")
    print(f"MX Found: {response.mx_found}")
    print(f"Check Status: {response.check_status}")
except Exception as e:
    print(f"Error: {e}")

File: typescript/src/email_validation_test.ts
import { Configuration, ValidationApi } from '@onboardingbuddy/onboardingbuddy-client/dist/client';
import * as dotenv from 'dotenv'
dotenv.config()

const obAppKey = process.env.OB_APP_KEY || '';
const obApiKey = process.env.OB_API_KEY || '';
const obApiSecret = process.env.OB_API_SECRET || '';

//const client = new OnboardingBuddyClient(obAppKey, obApiKey, obApiSecret);
const config = new Configuration({
      baseOptions: {
        headers: {
            'ob-app-key': obAppKey,
            'ob-api-key': obApiKey,
            'ob-api-secret': obApiSecret,
        }
      },
    });
const validationApi = new ValidationApi(config);
    
// Email Validation
const emailValidation = validationApi.email({
  emailAddress: 'email@domain.com',
});

emailValidation
    .then(response => 
        { 
            console.log(`Email Address: ${response.data.emailAddress}`)
            console.log(`Email Status: ${response.data.emailStatus}`)
            console.log(`Free Email: ${response.data.freeEmail}`)
            console.log(`Domain: ${response.data.domain}`)
            console.log(`MX Record: ${response.data.mxRecord}`)
            console.log(`MX Found: ${response.data.mxFound}`)
            console.log(`Check Status: ${response.data.checkStatus}`)
        })
    .catch(error=> console.log(error.response.data));
